Shared/Samples/Set reference scale/SetReferenceScaleView.swift

```diff
@@ -0,0 +1,166 @@
+                                // when the "Set to Reference Scale" button is tapped.
+                                Task {
+                                    await mapViewProxy.setViewpointScale(scale)
+                                }
+                            }
+                            .presentationDetents([.fraction(0.5)])
+                            .frame(idealWidth: 320, idealHeight: 380)
+                        }
+                    }
+                }
+        }
+    }
+}
+
+private extension SetReferenceScaleView {
+    /// The view containing settings controls for a map.
+    struct MapSettingsView: View {
+        /// The map.
+        let map: Map
+        
+        /// A binding to the scale of the map.
+        let mapScale: Double
+        
+        /// The action to set the map scale, i.e, when the "Set to Reference Scale" button is pressed.
+        let setMapScaleAction: (_ scale: Double) -> Void
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss
+        
+        /// The reference scale option selected in the picker.
+        @State private var selectedReferenceScale: Double = 250_000
+        
+        /// The options for the reference scale picker
+        private let referenceScaleOptions: [Double] = [500_000, 250_000, 100_000, 50_000]
+        
+        var body: some View {
+            NavigationStack {
+                Form {
+                    Section {
+                        Picker("Reference Scale", selection: $selectedReferenceScale) {
+                            ForEach(referenceScaleOptions, id: \.self) { option in
+                                Text("1:\(option, format: .number.rounded(increment: 1))")
+                            }
+                        }
+                        .onChange(of: selectedReferenceScale) { map.referenceScale = $0 }
+                        
+                        NavigationLink("Layers") {
+                            Form {
+                                ForEach(map.operationalLayers as! [FeatureLayer], id: \.id) { layer in
+                                    ScalesSymbolsToggle(layer: layer)
+                                }
+                            }
+                            .navigationTitle("Layers")
+                            .navigationBarTitleDisplayMode(.inline)
+                            .toolbar {
+                                ToolbarItem(placement: .confirmationAction) {
+                                    Button("Done") { dismiss() }
+                                }
+                            }
+                        }
+                    } header: {
+                        Text("Reference Scale")
+                    } footer: {
+                        Text("Selected layers will scale according to the reference scale.")
+                    }
+                    
+                    Section("Map Scale") {
```

Comment:
I don't think "Map Scale" needs to be repeated in the header. Having a row with that label is sufficient.

```suggestion
                    Section {
```